File: Answer/2091.从数组中移除最大值和最小值.py
#
# @lc app=leetcode.cn id=2091 lang=python3
#
# [2091] 从数组中移除最大值和最小值
#

# @lc code=start
from typing import List


class Solution:
    def minimumDeletions(self, nums: List[int]) -> int:
        min_index = 0
        max_index = 0
        for i in range(1, len(nums)):
            if nums[i] < nums[min_index]:
                min_index = i
            if nums[i] > nums[max_index]:
                max_index = i
        if min_index > max_index:
            min_index, max_index = max_index, min_index
        n = len(nums)
        return min(max_index + 1, n - min_index, min_index + 1 + n - max_index)
# @lc code=end

File: Answer/2091.从数组中移除最大值和最小值.dart
/*
 * @lc app=leetcode.cn id=2091 lang=dart
 *
 * [2091] 从数组中移除最大值和最小值
 */

// @lc code=start
class Solution {
  int minimumDeletions(List<int> nums) {
    final n = nums.length;
    var minIndex = 0;
    var maxIndex = 0;
    for (var i = 1; i < n; i++) {
      if (nums[i] < nums[minIndex]) {
        minIndex = i;
      }
      if (nums[i] > nums[maxIndex]) {
        maxIndex = i;
      }
    }
    if (minIndex > maxIndex) {
      final temp = minIndex;
      minIndex = maxIndex;
      maxIndex = temp;
    }
    var ans = maxIndex + 1;
    if (n - minIndex < ans) {
      ans = n - minIndex;
    }
    if (minIndex + 1 + n - maxIndex < ans) {
      ans = minIndex + 1 + n - maxIndex;
    }
    return ans;
  }
}
// @lc code=end